invenio_collections/receivers.py

```diff
@@ -100,7 +100,9 @@ def __init__(self, app=None):
         :param app: Flask application.
         """
         app = app or current_app
-        if not app.config['COLLECTIONS_USE_PERCOLATOR']:
+        if app.config['COLLECTIONS_MATCHER']:
+            self.matcher = current_collections.matcher
```

Comment:
This might not be correct, because it makes this method require an application context when `COLLECTIONS_MATCHER` is set.